### PR TITLE
Updated code to GNU Radio 3.10

### PR DESCRIPTION
**Pull request by dkozel**
Hello. I just had a chance to quickly test this on an SMU-200 and it worked pretty well. I updated it to GNU Radio 3.10 and fixed one small bug where the "CLOCK" property in the file was not being set to the sample rate.

There is one decrease in functionality, GNU Radio changed it's audio sink and the function to automatically handle converting from host endianness to little-endian was removed. I brute force copied that code from GNU Radio 3.7 but didn't keep the check for host endianness so the block currently won't work if the host is big endian.

If you're interested in merging this update I can fix that and restore all the copyright notices correctly.

rs-waveforms-demo

### CODE REVIEW

**Comment by AndersKaloer** [comment]
Excellent, thanks a lot!
It would be great if you could fix the endianness issue, although I am not sure how important it is in practice. I guess it also would be fine to just support little endian hosts.

I would be happy to test on hardware, but I will not have access until after the holidays.

**Comment by dkozel** [comment]
I don't have a big-endian host to test with, but I'm reasonably confident that this solution handles the sample conversion if needed. Once C++20 becomes required for GNU Radio there's some additional simplification possible, though the first step really is to interleave the IQ into one array in memory, byte swap all the elements in one pass, and write the entire buffer to the file to allow the compiler more room for optimisation.

I don't have any hardware to test with, but the R&S ARB Toolkit reads the new file correctly.
image

**Comment by dkozel** [comment]
Commits are tidied, if you would like to to squash all these into one I'm happy to do that. Thanks for writing the module!

**Comment by nickoe** [comment]
Hi @dkozel

Thank you very much for updating this, it is very helpful and much appreciated.  I was wondering if there is a chance to upstream this to gnuradio if there is no licensing issues, I guess this would make it more useful over time and maybe someone would implement a source for end to end testing?

I just tried to build this on archlinux on my workstation as is, but I get a build error. 

I wonder why I see this in the cmake configure:
```
...
-- Could NOT find PythonLibs: Found unsuitable version "2.7.18", but required is exact version "3.11" (found /usr/lib/libpython2.7.so)
-- Extracting version information from git describe...
fatal: No names found, cannot describe anything.
...
```

I do have 3.11.. and not sure why git describe also fails.  I guess those are minor issues.  I can't easily test it now and R&S ARB Toolbox does not run for me in wine.

EDIT2: I updated my comment to reflect reality :D

File: MANIFEST.md
Comment (nickoe) [inline MANIFEST.md:1]: Maybe keep the old info from the manifest md?

File: MANIFEST.md [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
 author:
-  - Anders Kalør <anders@kaloer.com>
+  - Author Name <authors@email.address>
 copyright_owner:
-  - Anders Kalør <anders@kaloer.com>
-license: GPLv3
-
+  - Copyright Owner 1
+license:
+gr_supported_version: # Put a comma separated list of supported GR versions here
```
Comment (nickoe) [inline MANIFEST.md]: 3.10, 3.11?

File: README.md [hunk 2]
```diff
@@ -1,17 +1,17 @@
 # gr-rswaveforms
-GNURadio sink that creates Rohde & Schwarz SMU waveform files (.wv) for R&S signal generators (e.g. SMU, SMBV).
+
+A GNU Radio module for creating Rohde & Schwarz SMU waveform files (.wv) for Rohde & Schwarz signal generators (e.g. SMU, SMV).
```
Comment (nickoe) [inline README.md]: I am not familiar with the SMV, but I know the SMBV so I don't think it was a typo per se.

Comment (dkozel) [reply README.md]: Changed SMV back to SMBV. :+1:

**Comment by dkozel** [comment]
@nickoe Thanks for pointing out the Manifest file. I'd completely overlooked it. I've added that back in and squashed it into 00cfca3.

Implementing a source block seems quite straightforward and would be a nice addition, I agree.

I don't think this would be considered for upstreaming to the core, if anything the core is likely to be trimmed down significantly in the 4.0 release.  I am working with Kitware to significantly streamline the installation and management of OOT modules, so hopefully these become easier for people to find and use. There's also hopes for CI for all these OOT modules so authors can get feedback about when changes to GR or operating systems break things.

**Comment by AndersKaloer** [comment]
Thank you both, @dkozel and @nickoe, for implementing this!
I trust the R&S ARB Toolkit for now, but hope to be able to verify it on HW soon.

**Pull request merged**